### PR TITLE
update get_user_notification_preferences function [SDESK-7366]

### PR DESCRIPTION
**Pull request by devketanpro**

### CODE REVIEW

**Review by petrjasek — COMMENTED** [review]
can you add a test case? not sure what exactly it's fixing

**Comment by devketanpro** [comment]
previously, the function wasn’t searching for values inside the notifications field; it was always checking `user["user_preferences"]["notification_name"]`. I’ve now updated the function so that if a notification name is provided, it will check `user["user_preferences"]["notifications"]["notification_name"]...` instead.

File: tests/preferences/preference_tests.py [hunk 1]
```diff
@@ -161,16 +161,18 @@ def test_get_user_notification_preferences_with_email_prefix(self):
             username="foo",
             user_preferences={
                 "email:notification": {"enabled": True},
-                "email:notification:test": {"enabled": False},
+                "notifications": {"email:notification:test": {"email": False, "desktop": True}},
```
Comment (petrjasek) [inline tests/preferences/preference_tests.py:164]: that's not how it should be working, having `email` prefixed notification inside `notifications`

Comment (devketanpro) [reply tests/preferences/preference_tests.py:164]: currently, this is how it works

Comment (devketanpro) [reply tests/preferences/preference_tests.py:164]: currently we store it in this way :- `"notifications": {"email:notification:assignments": {"email": False, "desktop": True}},` right @thecalcc ?

Comment (thecalcc) [reply tests/preferences/preference_tests.py:164]: we store it with the notification id and then `email: boolean` & `desktop: boolean`

Comment (petrjasek) [reply tests/preferences/preference_tests.py:164]: atm we don't store it inside `notifications` right. once we do we shouldn't have there `email:notification` in the name. I only had there this code with `email:notification` prefix to be backwards compatible, but it shouldn't be used in future and definitely not inside `notifications` dict

**Comment by devketanpro** [comment]
don;t need to change anything in the core

**Pull request closed without merging**